tests/integration/dart/test/login_configuration_client_test.dart

```diff
@@ -0,0 +1,224 @@
+      allowListApi = loginConfigurationClient.getAllowListApi();
+      denyListApi = loginConfigurationClient.getDenyListApi();
+      configurationApi = loginConfigurationClient.getConfigurationApi();
+    });
+
+    group('Login Configurations', () {
+      test('Creates login configuration', () async {
+        final name = 'TestConfig';
+        final redirectUri = 'http://localhost:3000/api/auth/callback/affinidi';
+
+        final createLoginConfigurationInput = CreateLoginConfigurationInputBuilder()
+          ..name = name
+          ..redirectUris = ListBuilder<String>([redirectUri]);
+
+        final response = (await configurationApi.createLoginConfigurations(
+              createLoginConfigurationInput: createLoginConfigurationInput.build()));
+
+        expect(response.statusCode, 201);
+        expect(response.data!.configurationId, isNotEmpty);
+
+        configurationId = response?.data?.configurationId ?? '';
+      });
+
+      test('Reads login configurations', () async {
+        final configurations = (await configurationApi.listLoginConfigurations()).data?.configurations;
+        expect(configurations, isNotNull);
+
+        expect(configurations?.length, greaterThan(0));
+      }, skip: 'TODO: Fix API spec for LoginConfigurationObject -> null for non-nullable field `clientId`');
+
+      test('Updates login configuration', () async {
+        final String updatedName = 'UpdatedName';
+
+        final updateLoginConfigurationInput = UpdateLoginConfigurationInputBuilder()
+          ..name = updatedName;
+
+        final config = (await configurationApi.updateLoginConfigurationsById(
+          configurationId: configurationId,
+          updateLoginConfigurationInput: updateLoginConfigurationInput.build())).data;
+
+        expect(config, isNotNull);
+        expect(config?.name, equals(updatedName));
+      }, skip: 'TODO: Fix API spec for LoginConfigurationObject -> null for non-nullable field `clientId`');
+
+      // NOTE: Deserializing 'LoginConfigurationObject' failed due to: Tried to construct class "LoginConfigurationObject" with null for non-nullable field "clientId"
+      test('Reads login configuration', () async {
+        final config = (await configurationApi.getLoginConfigurationsById(configurationId: configurationId)).data;
+
+        expect(config, isNotNull);
+      }, skip: 'TODO: Fix API spec for LoginConfigurationObject -> null for non-nullable field `clientId`');
+
+      test('Deletes login configuration', () async {
+        if (configurationId.isNotEmpty) {
+          await configurationApi.deleteLoginConfigurationsById(configurationId: configurationId);
+
+          expectLater(
+            configurationApi.getLoginConfigurationsById(configurationId: configurationId),
+            throwsA(isA<DioException>().having((e) => e.response?.statusCode, 'status code', 404)),
+          );
+        }
+      });
+    });
+
+    group('User groups', () {
```

Comment:
Might be premature optimization, but given that we are calling multiple APIs sequentially in these tests, they may take some time to execute. Maybe we can check which tests take the longer and split them in different files so that they run in parallel. This group for example can be moved to a separate file.